lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -333,6 +360,28 @@ def cmd_debug(*args)
     end
   end
 
+  #
+  # Tab completion for the connect command
+  #
+  # @param str [String] the string currently being typed before tab was hit
+  # @param words [Array<String>] the previously completed words on the command line.  words is always
+  # at least 1 when tab completion has reached this stage since the command itself has been completed
+
+  def cmd_connect_tabs(str, words)
+    if words.length == 1
+      return @@connect_opts.option_keys.select do |opt|
+        opt.start_with?(str) && !words.include?(opt)
+      end
+    end
+
+    case words[-1]
+    when '-c', '--comm'
+      # Rex::Socket::Comm
```

Comment:
Separate PR: I believe this would be session IDs that extend `"Msf::Session::Comm`